config/config.go

```diff
@@ -158,15 +171,59 @@ const (
 	MetricsInflux MetricsType = "influx"
 )
 
-type Influx struct {
+type InfluxMetrics struct {
 	Host     string `mapstructure:"host"`
 	Database string `mapstructure:"database"`
 	Username string `mapstructure:"username"`
 	Password string `mapstructure:"password"`
+	Enabled  bool   `mapstructure:"enabled"`
+}
+
+func (influxMetricsConfig *InfluxMetrics) validateAndLogMetricsData() bool {
+	if influxMetricsConfig.Host == "" {
+		influxMetricsConfig.Enabled = false
+		log.Fatalf(`Despite being enabled, influx metrics came with no host info: config.metrics.influx.host = "".`)
+	}
+	if influxMetricsConfig.Database == "" {
+		influxMetricsConfig.Enabled = false
+		log.Fatalf(`Despite being enabled, influx metrics came with no database info: config.metrics.influx.database = "".`)
+	}
+	if influxMetricsConfig.Enabled {
+		log.Infof("config.metrics.influx.host: %s", influxMetricsConfig.Host)
+		log.Infof("config.metrics.influx.database: %s", influxMetricsConfig.Database)
+	}
+	return influxMetricsConfig.Enabled
+}
+
+type PrometheusMetrics struct {
+	Port             int    `mapstructure:"port"`
+	Namespace        string `mapstructure:"namespace"`
+	Subsystem        string `mapstructure:"subsystem"`
+	TimeoutMillisRaw int    `mapstructure:"timeout_ms"`
+	Enabled          bool   `mapstructure:"enabled"`
+}
+
+func (promMetricsConfig *PrometheusMetrics) validateAndLogMetricsData() bool {
```

Comment:
Same feedback as the influx version with the name and changing the struct `Enabled`.

Reply:
Modified to be more straightforward. Let me know what you think